source/interprocedural_analyses/taint/taintConfiguration.ml

```diff
@@ -1595,14 +1608,16 @@ let from_taint_model_paths taint_model_paths =
         |> File.content
         |> Result.of_option ~error:(Error.create ~path ~kind:FileRead)
       in
-      try content >>| Json.from_string >>| fun json -> path, json with
-      | Yojson.Json_error parse_error ->
-          Result.Error (Error.create ~path ~kind:(Error.InvalidJson parse_error))
+      try content >>| Json.json_of_string >>| fun json -> path, json with
+      | JsonParsing.JsonmException { message; location } ->
```

Comment:
Why are we returning a `JsonmException`?
Since we have our own library, we should not expose stuff from its underlying implementation, in case we change it later. We should define our own exception type - or better, use result types instead of raising exceptions.

Reply:
Okay, implemented a generic exception type :) I tried result types but couldn't get it working for some reason

Reply:
Could you try in another PR? There is no reason that wouldn't work.

Reply:
Got it working :)